### DISCUSSION TITLE
InMemoryDataset.copy() keep original indicies

### DISCUSSION THREAD

**Discussion by anthonysirico** [pyg-team/pytorch_geometric#5884] [Q&A] [answered]
Is there a way to keep the original index when using copy?

```
dataset = Dataset('test')
train = dataset.copy(idx=range(split))
test = dataset.copy(idx=range(split+1, len(dataset))
```

**Comment by rusty1s** [comment, marked as answer]
I guess the easiest way is to add a `graph_id` to each `data` object. This way, the initial indices are maintained. Would that work for you?

**Reply by anthonysirico** [reply]
Yes thank you.
